spec/controllers/queue_items_controller_spec.rb

```diff
@@ -107,28 +105,31 @@
     it "does not delete the queue item if the queue item is not in the current user's queue" do
       alice = Fabricate(:user)
       bob = Fabricate(:user)
-      session[:user_id] = alice.id
+      set_current_user(alice)
       queue_item = Fabricate(:queue_item, user: bob)
       delete :destroy, id: queue_item.id
       expect(QueueItem.count).to eq(1)
     end
     
-    it "redirects to the sign in page for unauthenticated users" do
-      delete :destroy, id: 3
-      expect(response).to redirect_to sign_in_path
+    it_behaves_like "requires sign in" do
+      let(:action) { delete :destroy, id: 3 }
     end
   end
   
   describe "POST update_queue" do
+    
+     it_behaves_like "requires sign in" do
+      let(:action) { post :update_queue, queue_items: [{id: 2, position: 3},
+        {id: 5, position: 2}] }
+    end
+    
     context "with valid inputs" do
-      alice = Fabricate(:user)
```

Comment:
This is the reason, why you used database cleaner! When you fabricate an object outside of the `it`, `before` or `let` blocks its going to be in the database, because RSpec runs in transactions.